guides/hack/11-built-in-types/35-enum-class.md

```diff
@@ -139,23 +193,23 @@ function show_name_from_names(\HH\MemberOf<Names, IHasName> $x): string {
 }
 
 function test2(): void {
-  show_name(new HasName('toto')); // error, this instance is not from the Names enum
-  show_name(Names::World); // no problem
+  // show_name_from_names(new HasName('toto')); // error, this instance is not from the Names enum
+  show_name_from_names(Names::World); // no problem
 }
 
 enum class OtherNames: IHasName {
   HasName Foo = new HasName('foo');
 }
 
 function test3(): void {
-  show_name(OtherNames::Foo); // error, expected Names but got OtherNames
+  // show_name_from_names(OtherNames::Foo); // error, expected Names but got OtherNames
 }
 ```.ini
 hhvm.hack.lang.enable_enum_classes=1
 ```
 
-
-As explained in the introduction, this feature also allows to write dependently typed code. Let’s consider the more involved code:
+## Motivation: Dependent Typing
+Our main goal with the enum class is to allow more expressivity and replace some generated / boiler plate code: associating enum constants to different subtypes of the base type provides a lightweight form of dependent types that can be used as type-safe abstraction to write generic validator/sanitizer/loggers boiler-plate code.
```

Comment:
- "The main goal of enum classes"
- this description feels very expert-focussed. E.g. what is a 'dependent type' ?

Reply:
For the examples, should the simple one come first? Also the purpose of the Box one is unclear - what advantages does `get(Box::AGE)`  have over `Box::AGE->data` ?

I think this is pre-existing:

> the return value of the get function depends on which constant is passed as an input.

I think 'the return type' is what makes this interesting?